tests/Silverback.Tests.Performance/KafkaKeyHelperBenchmark.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System.Diagnostics.CodeAnalysis;
+using BenchmarkDotNet.Attributes;
+using Silverback.Messaging.Messages;
+
+namespace Silverback.Tests.Performance
+{
+    // Original
+//    |                          Method |       Mean |      Error |     StdDev |     Median |   Gen 0 | Gen 1 | Gen 2 | Allocated |
+//    |-------------------------------- |-----------:|-----------:|-----------:|-----------:|--------:|------:|------:|----------:|
+//    |           MessagesWithSimpleKey | 256.076 us | 10.2684 us | 28.9623 us | 248.838 us | 20.0000 |     - |     - |     49 KB |
+//    | MessagesWithSimpleKeyButNoValue | 255.138 us |  8.1419 us | 22.8308 us | 253.428 us | 20.0000 |     - |     - |     49 KB |
+//    |         MessagesWithCombinedKey | 416.836 us | 14.1230 us | 39.6023 us | 406.556 us | 40.0000 |     - |     - |    100 KB |
+//    |              MessagesWithoutKey |   1.863 us |  0.1089 us |  0.3107 us |   1.777 us |       - |     - |     - |      2 KB |
+
+// After fast exit for dict count 1 and 0
+
+// |                          Method |       Mean |      Error |      StdDev |     Median |   Gen 0 | Gen 1 | Gen 2 | Allocated |
+// |-------------------------------- |-----------:|-----------:|------------:|-----------:|--------:|------:|------:|----------:|
+// |           MessagesWithSimpleKey | 257.561 us | 10.8819 us |  31.7429 us | 253.345 us | 20.0000 |     - |     - |     45 KB |
+// | MessagesWithSimpleKeyButNoValue | 286.306 us | 16.5246 us |  48.2031 us | 272.922 us | 20.0000 |     - |     - |     45 KB |
+// |         MessagesWithCombinedKey | 542.203 us | 41.2967 us | 121.7641 us | 501.106 us | 40.0000 |     - |     - |    100 KB |
+// |              MessagesWithoutKey |   1.943 us |  0.1667 us |   0.4889 us |   1.888 us |       - |     - |     - |      2 KB |
+
+// After introduce the cache, eliminate allocation closure for message, unnecessary call to "All" and string eliminate string allocs
+
+// |                          Method |         Mean |       Error |       StdDev |       Median |   Gen 0 | Gen 1 | Gen 2 | Allocated |
+// |-------------------------------- |-------------:|------------:|-------------:|-------------:|--------:|------:|------:|----------:|
+// |           MessagesWithSimpleKey |  26,775.8 ns | 2,578.15 ns |  7,561.27 ns |  28,802.0 ns |       - |     - |     - |         - |
+// | MessagesWithSimpleKeyButNoValue |  18,486.7 ns | 1,852.09 ns |  5,373.26 ns |  15,380.0 ns |       - |     - |     - |         - |

```

Comment:
Don't look at this. Made a mistake somewhere. It can be the case that there is no allocation.